series/series_linux.go

```diff
@@ -125,22 +149,61 @@ func updateDistroInfo() error {
 			}
 			foundPrecise = true
 		}
+		// we support non-LTS releases after the current release, everything
+		// before the current release that is not an LTS is not supported.
+		if !nonLTSSupported && series == "bionic" {
+			nonLTSSupported = true
+		}
 
 		releaseDate, err := time.Parse("2006-01-02", release)
 		if err != nil {
 			// Ignore lines with invalid release dates.
 			continue
 		}
 
+		eolDate, err := time.Parse("2006-01-02", eol)
+		if err != nil {
+			// we should add 5 years to the release date in case of an error
+			// parsing the eol date.
+			eolDate = releaseDate.Add(5 * year)
+		}
+
+		eolESMDate, err := time.Parse("2006-01-02", eolESM)
+		if err != nil {
+			// fall back to the eolDate if none is provided in the csv.
+			eolESMDate = eolDate
+		}
+
 		// The numeric version may contain a LTS moniker so strip that out.
 		trimmedVersion := strings.TrimSuffix(version, " LTS")
 		seriesVersions[series] = trimmedVersion
-		ubuntuSeries[series] = trimmedVersion
-		if trimmedVersion != version && !now.Before(releaseDate) {
+
+		var ltsRelease bool
+		if strings.HasSuffix(version, " LTS") && !now.Before(releaseDate) {
 			// We only record that a series is LTS if its release
 			// date has passed. This allows the series to be tested
 			// pre-release, without affecting default series.
-			ubuntuLTS[series] = true
+			ltsRelease = true
+		}
+
+		// work out if the series is supported or if the extended security
+		// maintenance is supported from the following release cycle
+		// documentation https://www.ubuntu.com/about/release-cycle
+		var supported bool
+		if (ltsRelease || nonLTSSupported) && now.After(releaseDate) && now.Before(eolDate) {
```

Comment:
I guess this might catch the case of the release cycle and what we do/don't support. But then why would we need to track nonLTSSupported?

Reply:
Exactly.